src/hocon_schema.erl

```diff
@@ -67,7 +67,7 @@
 -callback translations() -> [name()].
 -callback translation(name()) -> [translation()].
 -callback validations() -> [validation()].
--callback desc(name()) -> iodata() | undefined.
+-callback desc(name()) -> desc().
```

Comment:
I think there are several usages in EMQX that return `undefined` as well. :thinking: 

```suggestion
-callback desc(name()) -> desc() | undefined.
```

Reply:
Ops... fixed (curiously dialyzer didn't complain about it locally...)